sympy/core/function.py

```diff
@@ -716,17 +716,15 @@ def _eval_nseries(self, x, n, logx):
             return e1.nseries(x, n=n, logx=logx)
         arg = self.args[0]
         l = []
+        i = 0
         g = None
-        # try to predict a number of terms needed
-        nterms = n + 2
-        cf = Order(arg.as_leading_term(x), x).getn()
-        if cf != 0:
-            nterms = int(nterms / cf)
-        for i in range(nterms):
+        while len(l) <= n:
```

Comment:
This is inefficient. `n` is not the number of terms but the order (exponent of the order term). For `cos(x**6).series(x, 0, 14)`  this loop would compute 14 non-zero terms though only 2 are needed (3 if also the linear term `0*arg` is counted).

Reply:
Okay I will done the changes suggested by you.